src/main/java/com/google/step/YTLounge/data/YoutubeService.java

```diff
@@ -0,0 +1,51 @@
+package com.google.step.YTLounge.data;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Scanner;
+
+public class YoutubeService {
+  private static final JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+  public YoutubeService() {}
+
+  /**
+   * Build and return an authorized API client service.
+   *
+   * @return an authorized API client service
+   * @throws GeneralSecurityException, IOException
+   */
+  public static YouTube getService() throws GeneralSecurityException, IOException {
+    String key = YoutubeService.getAPIKey();
+    final YouTubeRequestInitializer keyInitializer = new YouTubeRequestInitializer(key);
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    return new YouTube.Builder(httpTransport, jsonFactory, null)
+        .setApplicationName("YouTube Lounge")
+        .setYouTubeRequestInitializer(keyInitializer)
+        .build();
+  }
+
+  /** Locate the necessary API key to access needed data */
+  private static String getAPIKey() {
+    try {
+      File secretFile = new File("dataSecret.txt");
+      Scanner myReader = new Scanner(secretFile);
+      if (myReader.hasNextLine()) {
+        String data = myReader.nextLine();
+        return data;
+      }
+      myReader.close();
```

Comment:
reader.close() wouldn't be triggered right? if you return the data.

Reply:
Yeah, it's just in the case that the dataSecret.txt file is gone which protects from having an illegalStateException and closing any open streams